Remove RemoteMediaRecorderPrivateWriter and it's Manager

jyavenard opened the pull request:
#### fe16843ed00fdb78d831082d5bb96ff1e01caf22
<pre>
Remove RemoteMediaRecorderPrivateWriter and it&apos;s Manager
<a href="https://bugs.webkit.org/show_bug.cgi?id=295449">https://bugs.webkit.org/show_bug.cgi?id=295449</a>
<a href="https://rdar.apple.com/155011183">rdar://155011183</a>

Reviewed by Youenn Fablet.

They are no longer in use as on all supported platforms, AVAssetWriter can now work in process.

No change in observable behaviour.

* Source/WebCore/loader/LocalFrameLoaderClient.h: Fix missing header.
(WebCore::MediaStrategy::createMediaRecorderPrivateWriter const): Deleted.
* Source/WebCore/platform/MediaStrategy.h: Remove virtual methods no longer needed.
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateWriter.cpp: Remove use of MediaStrategy.
(WebCore::MediaRecorderPrivateWriter::create):
(WebKit::GPUConnectionToWebProcess::remoteMediaRecorderPrivateWriterManager): Deleted.
(WebKit::GPUConnectionToWebProcess::protectedRemoteMediaRecorderPrivateWriterManager): Deleted.
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.cpp: Removed.
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.h: Removed.
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.messages.in: Removed.
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in: Remove serialisation of removed type and identifier.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriter.cpp: Removed.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriter.h: Removed.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriterIdentifier.h: Removed.
(WebKit::WebMediaStrategy::createMediaRecorderPrivateWriter const): Deleted.
* Source/WebKit/WebProcess/Inspector/WebInspectorUIExtensionController.cpp: Add missing header
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.h: Add missing header

Canonical link: <a href="https://commits.webkit.org/297051@main">https://commits.webkit.org/297051@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b8d681402576bfdfddc6b17be99c525c1000490

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110440 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30099 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20532 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116465 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60694 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112403 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30778 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38686 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83985 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113388 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24564 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99445 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64426 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/109872 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23927 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17584 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60261 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/102931 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93948 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17640 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119256 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/108994 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37480 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27811 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92952 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37858 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95713 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92775 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23628 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37781 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15510 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33426 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37375 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42846 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/133269 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37037 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36017 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40377 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38745 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -8015,7 +8004,7 @@
 		CD8252E025D4918400862FD8 /* RemoteRemoteCommandListenerMessageReceiver.cpp */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.cpp.cpp; path = RemoteRemoteCommandListenerMessageReceiver.cpp; sourceTree = "<group>"; };
 		CD8252E125D4918500862FD8 /* RemoteRemoteCommandListenerMessages.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = RemoteRemoteCommandListenerMessages.h; sourceTree = "<group>"; };
-		CD95493526159004008372D9 /* .dylib */ = {isa = PBXFileReference; explicitFileType = "compiled.mach-o.dylib"; includeInIndex = 0; path = .dylib; sourceTree = BUILT_PRODUCTS_DIR; };
+		CD95493526159004008372D9 /* libWebKitSwift.dylib */ = {isa = PBXFileReference; explicitFileType = "compiled.mach-o.dylib"; includeInIndex = 0; path = libWebKitSwift.dylib; sourceTree = BUILT_PRODUCTS_DIR; };
```
Seems like a strange change, I guess it is done automatically so is fine

youennf commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -2591,7 +2589,6 @@
 		F4E090952D889D6A00322226 /* WKIdentityDocumentPresentmentDelegate.h in Headers */ = {isa = PBXBuildFile; fileRef = F4E090942D889CB800322226 /* WKIdentityDocumentPresentmentDelegate.h */; };
 		F4E090992D88AAC300322226 /* WKIdentityDocumentPresentmentRawRequest.h in Headers */ = {isa = PBXBuildFile; fileRef = F4E090962D88A66F00322226 /* WKIdentityDocumentPresentmentRawRequest.h */; };
 		F4E28A362C923814008120DD /* ScriptTrackingPrivacyFilter.h in Headers */ = {isa = PBXBuildFile; fileRef = F4E28A352C923814008120DD /* ScriptTrackingPrivacyFilter.h */; };
-		F4E44EB72DCD33E800A304C4 /* ISO18013MobileDocumentRequest+Extras.swift in Sources */ = {isa = PBXBuildFile; fileRef = F4E44EB62DCD33E800A304C4 /* ISO18013MobileDocumentRequest+Extras.swift */; };
```
Isn't this odd?

jyavenard replied:
Yes,no idea on how/why this occurred

jyavenard committed:
Remove RemoteMediaRecorderPrivateWriter and it's Manager
https://bugs.webkit.org/show_bug.cgi?id=295449
rdar://155011183

Reviewed by Youenn Fablet.

They are no longer in use as on all supported platforms, AVAssetWriter can now work in process.

No change in observable behaviour.

* Source/WebCore/loader/LocalFrameLoaderClient.h: Fix missing header.
(WebCore::MediaStrategy::createMediaRecorderPrivateWriter const): Deleted.
* Source/WebCore/platform/MediaStrategy.h: Remove virtual methods no longer needed.
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateWriter.cpp: Remove use of MediaStrategy.
(WebCore::MediaRecorderPrivateWriter::create):
(WebKit::GPUConnectionToWebProcess::remoteMediaRecorderPrivateWriterManager): Deleted.
(WebKit::GPUConnectionToWebProcess::protectedRemoteMediaRecorderPrivateWriterManager): Deleted.
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.cpp: Removed.
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.h: Removed.
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.messages.in: Removed.
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in: Remove serialisation of removed type and identifier.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriter.cpp: Removed.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriter.h: Removed.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriterIdentifier.h: Removed.
(WebKit::WebMediaStrategy::createMediaRecorderPrivateWriter const): Deleted.
* Source/WebKit/WebProcess/Inspector/WebInspectorUIExtensionController.cpp: Add missing header
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.h: Add missing header

Canonical link: https://commits.webkit.org/297051@main
